pkg/deploy/lattice/service_synthesizer.go

```diff
@@ -65,6 +70,14 @@ func (s *serviceSynthesizer) Synthesize(ctx context.Context) error {
 			s.latticeDataStore.AddLatticeService(resService.Spec.Name, resService.Spec.Namespace,
 				serviceStatus.ServiceARN, serviceStatus.ServiceID, serviceStatus.ServiceDNS)
 
+			glog.V(6).Infof("Creating Service DNS")
+			resService.Status = &serviceStatus
+			err = s.dnsEndpointManager.Create(ctx, resService)
+			if err != nil {
+				glog.V(2).Infof("Failed creating service DNS: %v", err)
+				return err
+			}
+
```

Comment:
Do we need to do`s.latticeDataStore.AddLatticeService()` at last, after  s.dnsEndpointManager.Create() finished? i.e., if `s.dnsEndpointManager.Create()` failed we should not do `s.latticeDataStore.AddLatticeService()`?

Reply:
dnsEndpoint is not a requirement for a service to function so I think we can still unblock other logic that uses it. I think latticeDataStore should represent what controller actually has in Lattice.